api/krusty/mergeenvfrom_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+// When patching, `envFrom` should merge the list instead of replacing it.
+func TestMergeEnvFrom(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteF("/app/base/deployment.yaml", `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  template:
+    spec:
+      containers:
+        - name: nginx
+          image: image1
+          envFrom:
+            - configMapRef:
+                name: some-config
+            - configMapRef:
+                name: more-config
+`)
+	th.WriteK("/app/base", `
+resources:
+- deployment.yaml
+
+patchesStrategicMerge:
+- |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx
+  spec:
+    template:
+      spec:
+        containers:
+          - name: nginx
+            envFrom:
+              - configMapRef:
+                  name: another-config
+`)
+	m := th.Run("/app/base", th.MakeDefaultOptions())
+	th.AssertActualEqualsExpected(m, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  template:
+    spec:
+      containers:
+      - envFrom:
+        - configMapRef:
+            name: another-config
+        image: image1
+        name: nginx
+`)
+}
```

Comment:
Please add the following to the test, see if if does the job you want
```
func TestMergeEnvFromViaJsonInline(t *testing.T) {
	th := kusttest_test.MakeHarness(t)
	th.WriteF("app/deployment.yaml", `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  template:
    spec:
      containers:
      - name: nginx
        image: image1
        envFrom:
        - configMapRef:
            name: some-config
        - configMapRef:
            name: more-config
`)
	th.WriteK("app", `
resources:
- deployment.yaml
patches:
- target:
    kind: Deployment
    name: nginx
  patch: |-
    - op: add
      path: /spec/template/spec/containers/0/envFrom/-
      value:
        configMapRef:
          name: another-config
`)
	m := th.Run("app", th.MakeDefaultOptions())
	th.AssertActualEqualsExpected(m, `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  template:
    spec:
      containers:
      - envFrom:
        - configMapRef:
            name: some-config
        - configMapRef:
            name: more-config
        - configMapRef:
            name: another-config
        image: image1
        name: nginx
`)
}

```

Reply:
That does the trick 🙂